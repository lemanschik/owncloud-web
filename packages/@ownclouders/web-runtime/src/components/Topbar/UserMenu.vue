<template>
  <nav v-if="userId" :aria-label="$gettext('Account menu')">
    <oc-button
      id="_userMenuButton"
      ref="menuButton"
      class="oc-topbar-personal"
      appearance="raw"
      :aria-label="$gettext('User Menu')"
    >
      <avatar-image
        class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle"
        :width="32"
        :userid="userId"
        :user-name="user.displayname"
      />
    </oc-button>
    <oc-drop
      ref="menu"
      drop-id="account-info-container"
      toggle="#_userMenuButton"
      mode="click"
      close-on-click
      padding-size="small"
    >
      <oc-list class="user-menu-list">
        <li>
          <oc-button
            id="oc-topbar-account-manage"
            type="router-link"
            :to="{ path: '/account' }"
            appearance="raw"
          >
            <avatar-image :width="32" :userid="userId" :user-name="user.displayname" />
            <span class="profile-info-wrapper" :class="{ 'oc-py-xs': !user.email }">
              <span class="oc-display-block" v-text="user.displayname" />
              <span v-if="user.email" class="oc-text-small" v-text="user.email" />
            </span>
          </oc-button>
        </li>
        <li v-for="(app, index) in applicationsList" :key="`user-menu-${index}`">
          <oc-button v-if="app.url" type="a" appearance="raw" :target="app.target" :href="app.url">
            <oc-icon :name="app.icon" class="oc-p-xs" />
            <span v-text="$gettext(app.title)" />
          </oc-button>
          <oc-button v-else type="router-link" appearance="raw" :to="{ path: app.path }">
            <oc-icon :name="app.icon" class="oc-p-xs" />
            <span v-text="$gettext(app.title)" />
          </oc-button>
        </li>
        <li>
          <oc-button id="oc-topbar-account-logout" appearance="raw" @click="logout">
            <oc-icon name="logout-box-r" fill-type="line" class="oc-p-xs" />
            <span v-text="$gettext('Log out')" />
          </oc-button>
        </li>
        <li v-if="quotaEnabled" class="storage-wrapper oc-pl-s">
          <oc-icon name="cloud" fill-type="line" class="oc-p-xs" />
          <div class="storage-wrapper-text">
            <p class="oc-my-rm">
              <span class="oc-display-block" v-text="personalStorageLabel" />
              <span class="oc-text-small" v-text="personalStorageDetailsLabel" />
            </p>
            <oc-progress
              v-if="limitedPersonalStorage"
              :value="quotaUsagePercent"
              :max="100"
              size="small"
              :variation="quotaProgressVariant"
            />
          </div>
        </li>
      </oc-list>
    </oc-drop>
  </nav>
</template>

<script lang="ts">
import { defineComponent, PropType, ComponentPublicInstance } from 'vue'
import { mapGetters, mapState } from 'vuex'
import filesize from 'filesize'
import isNil from 'lodash-es/isNil'
import { authService } from '../../services/auth'
import { useCapabilitySpacesEnabled } from 'web-pkg/src/composables'
import { OcDrop } from 'design-system/src/components'

export default defineComponent({
  props: {
    applicationsList: {
      type: Array as PropType<any>,
      required: false,
      default: () => []
    }
  },
  setup() {
    return {
      hasSpaces: useCapabilitySpacesEnabled()
    }
  },
  computed: {
    ...mapGetters(['quota', 'user']),
    ...mapGetters({ legacyQuota: 'quota' }),
    ...mapState('runtime/spaces', ['spaces']),

    quota() {
      return this.hasSpaces
        ? this.spaces.find((s) => s.driveType === 'personal')?.spaceQuota
        : this.legacyQuota
    },
    useLegacyQuota() {
      return !this.hasSpaces
    },

    userId() {
      return this.user.username || this.user.id
    },
    personalStorageLabel() {
      if (!this.limitedPersonalStorage) {
        return this.$gettext('Personal storage')
      }
      return this.$gettextInterpolate(this.$gettext('Personal storage (%{percentage}% used)'), {
        percentage: this.quotaUsagePercent || 0
      })
    },
    personalStorageDetailsLabel() {
      const total = this.quota.definition === 'none' ? 0 : this.quota.total || 0
      const used = this.quota.used || 0
      return this.$gettextInterpolate(
        total ? this.$gettext('%{used} of %{total} used') : this.$gettext('%{used} used'),
        {
          used: filesize(used),
          total: filesize(total)
        }
      )
    },
    limitedPersonalStorage() {
      if (!this.useLegacyQuota) {
        return this.quota.total !== 0
      }

      return !isNil(this.quota.relative) && this.quota.definition !== 'none'
    },
    quotaEnabled() {
      return !!this.quota
    },
    quotaUsagePercent() {
      return this.useLegacyQuota
        ? this.quota.relative
        : parseFloat(((this.quota.used / this.quota.total) * 100).toFixed(2))
    },

    quotaProgressVariant() {
      if ((this.quotaUsagePercent || 0) < 80) {
        return 'primary'
      }
      if ((this.quotaUsagePercent || 0) < 90) {
        return 'warning'
      }
      return 'danger'
    }
  },
  mounted() {
    ;(this.$refs.menu as InstanceType<typeof OcDrop>)?.tippy?.setProps({
      onHidden: () => (this.$refs.menuButton as ComponentPublicInstance).$el.focus(),
      onShown: () =>
        (this.$refs.menu as ComponentPublicInstance).$el.querySelector('a:first-of-type').focus()
    })
  },
  methods: {
    logout() {
      authService.logoutUser()
    }
  }
})
</script>

<style lang="scss" scoped>
.user-menu-list li {
  align-items: center;
  display: flex;
  margin: var(--oc-space-xsmall) 0;

  &:first-child {
    margin-top: 0;
  }
  &:last-child {
    margin-bottom: 0;
  }

  a,
  button {
    gap: var(--oc-space-medium);
    justify-content: flex-start;
    min-height: 3rem;
    padding-left: var(--oc-space-small);
    width: 100%;

    &:focus,
    &:hover {
      background-color: var(--oc-color-background-hover);
      color: var(--oc-color-swatch-passive-default);
      text-decoration: none;
    }

    .profile-info-wrapper {
      text-align: left;
      word-break: break-all;
      line-height: initial;
    }
  }

  &.storage-wrapper {
    gap: var(--oc-space-medium);
    min-height: 3rem;

    .storage-wrapper-text {
      align-self: flex-end;
      display: inline-block;
    }
  }
}
</style>
