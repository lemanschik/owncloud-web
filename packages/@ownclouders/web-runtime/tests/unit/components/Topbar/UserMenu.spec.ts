import UserMenu from 'web-runtime/src/components/Topbar/UserMenu.vue'
import {
  createStore,
  defaultPlugins,
  defaultStubs,
  mount,
  defaultStoreMockOptions
} from 'web-test-helpers'

const totalQuota = 1000
const basicQuota = 300
const warningQuota = 810
const dangerQuota = 910

const basicRelativeQuota = (basicQuota / totalQuota) * 100
const warningRelativeQuota = (warningQuota / totalQuota) * 100
const dangerRelativeQuota = (dangerQuota / totalQuota) * 100

const noEmail = ''
const email = 'test@test.de'

describe('User Menu component', () => {
  describe('when quota and no email is set', () => {
    it('renders a navigation without email', () => {
      const wrapper = getMountedWrapper(
        { used: basicQuota, total: totalQuota, relative: basicRelativeQuota },
        noEmail
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when no quota and email is set', () => {
    it('the user menu does not contain a quota', () => {
      const wrapper = getMountedWrapper(null, email)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when no quota and no email is set', () => {
    it('the user menu does not contain a quota', () => {
      const wrapper = getMountedWrapper(null, noEmail)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is below 80%', () => {
    it('renders a primary quota progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: basicQuota,
          total: totalQuota,
          relative: basicRelativeQuota,
          definition: 'default'
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is above 80% and below 90%', () => {
    it('renders a warning quota progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: warningQuota,
          total: totalQuota,
          relative: warningRelativeQuota,
          definition: 'default'
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is above 90%', () => {
    it('renders a danger quota progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: dangerQuota,
          total: totalQuota,
          relative: dangerRelativeQuota,
          definition: 'default'
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is unlimited', () => {
    it('renders no percentag of total and no progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: basicQuota,
          definition: 'default'
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('when quota is not defined', () => {
    it('renders no percentag of total and no progress bar', () => {
      const wrapper = getMountedWrapper(
        {
          used: dangerQuota,
          total: totalQuota,
          relative: dangerRelativeQuota,
          definition: 'none'
        },
        email
      )
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})

const getMountedWrapper = (quota, userEmail) => {
  const storeOptions = defaultStoreMockOptions
  storeOptions.getters.quota.mockImplementation(() => quota)
  storeOptions.getters.user.mockImplementation(() => ({
    id: 'einstein',
    username: 'einstein',
    userDisplayName: 'Albert Einstein',
    userEmail
  }))
  const store = createStore(storeOptions)
  return mount(UserMenu, {
    props: {
      applicationsList: [
        {
          icon: 'application',
          path: '/settings',
          title: 'Settings'
        }
      ]
    },
    global: {
      renderStubDefaultSlot: true,
      plugins: [...defaultPlugins(), store],
      stubs: {
        ...defaultStubs,
        'oc-button': true,
        'oc-drop': true,
        'oc-list': true,
        'avatar-image': true,
        'oc-icon': true,
        'oc-progress': true
      }
    }
  })
}
