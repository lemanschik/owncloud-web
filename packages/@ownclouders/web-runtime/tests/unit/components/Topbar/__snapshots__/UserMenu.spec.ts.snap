// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Menu component when no quota and email is set the user menu does not contain a quota 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <!--v-if-->
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when no quota and no email is set the user menu does not contain a quota 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <!--v-if-->
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when quota and no email is set renders a navigation without email 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <li class="storage-wrapper oc-pl-s">
        <oc-icon-stub class="oc-p-xs" fill-type="line" name="cloud"></oc-icon-stub>
        <div class="storage-wrapper-text">
          <p class="oc-my-rm">
            <span class="oc-display-block">Personal storage (30% used)</span>
            <span class="oc-text-small">300 B of 1 kB used</span>
          </p>
          <oc-progress-stub max="100" size="small" value="30" variation="primary"></oc-progress-stub>
        </div>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when quota is above 80% and below 90% renders a warning quota progress bar 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <li class="storage-wrapper oc-pl-s">
        <oc-icon-stub class="oc-p-xs" fill-type="line" name="cloud"></oc-icon-stub>
        <div class="storage-wrapper-text">
          <p class="oc-my-rm">
            <span class="oc-display-block">Personal storage (81% used)</span>
            <span class="oc-text-small">810 B of 1 kB used</span>
          </p>
          <oc-progress-stub max="100" size="small" value="81" variation="warning"></oc-progress-stub>
        </div>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when quota is above 90% renders a danger quota progress bar 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <li class="storage-wrapper oc-pl-s">
        <oc-icon-stub class="oc-p-xs" fill-type="line" name="cloud"></oc-icon-stub>
        <div class="storage-wrapper-text">
          <p class="oc-my-rm">
            <span class="oc-display-block">Personal storage (91% used)</span>
            <span class="oc-text-small">910 B of 1 kB used</span>
          </p>
          <oc-progress-stub max="100" size="small" value="91" variation="danger"></oc-progress-stub>
        </div>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when quota is below 80% renders a primary quota progress bar 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <li class="storage-wrapper oc-pl-s">
        <oc-icon-stub class="oc-p-xs" fill-type="line" name="cloud"></oc-icon-stub>
        <div class="storage-wrapper-text">
          <p class="oc-my-rm">
            <span class="oc-display-block">Personal storage (30% used)</span>
            <span class="oc-text-small">300 B of 1 kB used</span>
          </p>
          <oc-progress-stub max="100" size="small" value="30" variation="primary"></oc-progress-stub>
        </div>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when quota is not defined renders no percentag of total and no progress bar 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <li class="storage-wrapper oc-pl-s">
        <oc-icon-stub class="oc-p-xs" fill-type="line" name="cloud"></oc-icon-stub>
        <div class="storage-wrapper-text">
          <p class="oc-my-rm">
            <span class="oc-display-block">Personal storage</span>
            <span class="oc-text-small">910 B used</span>
          </p>
          <!--v-if-->
        </div>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;

exports[`User Menu component when quota is unlimited renders no percentag of total and no progress bar 1`] = `
<nav aria-label="Account menu">
  <oc-button-stub appearance="raw" aria-label="User Menu" class="oc-topbar-personal" id="_userMenuButton">
    <avatar-image-stub class="oc-topbar-personal-avatar oc-flex-inline oc-flex-center oc-flex-middle" userid="einstein" width="32"></avatar-image-stub>
  </oc-button-stub>
  <oc-drop-stub close-on-click="" drop-id="account-info-container" mode="click" padding-size="small" toggle="#_userMenuButton">
    <oc-list-stub class="user-menu-list">
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-manage" to="[object Object]" type="router-link">
          <avatar-image-stub userid="einstein" width="32"></avatar-image-stub>
          <span class="profile-info-wrapper oc-py-xs">
            <span class="oc-display-block"></span>
            <!--v-if-->
          </span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" to="[object Object]" type="router-link">
          <oc-icon-stub class="oc-p-xs" name="application"></oc-icon-stub>
          <span>Settings</span>
        </oc-button-stub>
      </li>
      <li>
        <oc-button-stub appearance="raw" id="oc-topbar-account-logout">
          <oc-icon-stub class="oc-p-xs" fill-type="line" name="logout-box-r"></oc-icon-stub>
          <span>Log out</span>
        </oc-button-stub>
      </li>
      <li class="storage-wrapper oc-pl-s">
        <oc-icon-stub class="oc-p-xs" fill-type="line" name="cloud"></oc-icon-stub>
        <div class="storage-wrapper-text">
          <p class="oc-my-rm">
            <span class="oc-display-block">Personal storage</span>
            <span class="oc-text-small">300 B used</span>
          </p>
          <!--v-if-->
        </div>
      </li>
    </oc-list-stub>
  </oc-drop-stub>
</nav>
`;
